shell/browser/native_window.cc

```diff
@@ -685,9 +686,28 @@ void NativeWindow::NotifyWindowMessage(UINT message,
 }
 #endif
 
-void NativeWindow::UpdateDraggableRegions(
-    const std::vector<mojom::DraggableRegionPtr>& regions) {
-  draggable_region_ = DraggableRegionsToSkRegion(regions);
+int NativeWindow::NonClientHitTest(const gfx::Point& point) {
+  for (auto* provider : draggable_region_providers_) {
+    int hit = provider->NonClientHitTest(point);
+    if (hit != HTNOWHERE)
+      return hit;
+  }
+  return HTNOWHERE;
+}
+
+void NativeWindow::AddDraggableRegionProvider(
+    DraggableRegionProvider* provider) {
+  if (std::find(draggable_region_providers_.begin(),
```

Comment:
Maybe just use `std::set` for `draggable_region_providers_` to avoid duplicate items?

Reply:
The reason I chose `std::list` is to preserve ordering. I'm not sure if this really matters, but to be safe, I wanted it to be that the draggable region providers were checked in the order they were inserted.